torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -202,6 +202,26 @@ template std::shared_ptr<OwnerRRef<IValue>> RRefContext::createOwnerRRef<
 template std::shared_ptr<OwnerRRef<py::object>> RRefContext::createOwnerRRef<
     py::object>();
 
+template <typename T>
+std::shared_ptr<OwnerRRef<T>> RRefContext::getOwnerRRef(const RRefId& rrefId) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  const auto iter = owners_.find(rrefId);
+  if (iter == owners_.end()) {
+    // Scenario (1) RRef is used before it is created
+    ownerCV_.wait(lock, [&] { return owners_.find(rrefId) != owners_.end(); });
+    return std::static_pointer_cast<OwnerRRef<T>>(owners_[rrefId]);
+  } else {
+    // Scenario (2) retrieving an existing RRef
+    return std::static_pointer_cast<OwnerRRef<T>>(iter->second);
```

Comment:
nit, we can get rid of the else branch and just return at the end.

Reply:
We could get rid of the else branch, though the existing code is probably marginally more efficient, since it avoids an extra table lookup.

Reply:
sounds good, let's keep it :)